qiskit/qpy/__init__.py

```diff
@@ -322,6 +342,139 @@
+ParameterExpression.
+
+The first change is the :ref:`qpy_param_expr_v3` payload for the ``expr_size`` bytes following
+the header is now comprised of an array of ``PARAM_EXPR_ELEM_V13`` structs. The intent is for the
+array of elements to be read one struct at a time that describe the calls to make to reconstruct
+the :class:`.ParameterExpression`.
+
+PARAM_EXPR_ELEM_V13
+~~~~~~~~~~~~~~~~~~~
+
+The struct format is defined as:
+
+.. code-block:: c
+
+    struct {
+        unsigned char op_code;
+        char lhs_type;
+        char lhs[16];
+        char rhs_type;
+        char rhs[16];
+    } PARAM_EXPR_ELEM_V13;
+
+The ``op_code`` field is used to define the operation used for the operation
+added to the :class:`.ParameterExpression`. The value can be:
+
+.. list-table:: PARAM_EXPR_ELEM_V13 op code values
+   :header-rows: 1
+
+   * - ``op_code``
+     - :class:`.ParameterExpression` method
+   * - 0
+     - :meth:`~.ParameterExpression.__add__`
+   * - 1
+     - :meth:`~.ParameterExpression.__sub__`
+   * - 2
+     - :meth:`~.ParameterExpression.__mul__`
+   * - 3
+     - :meth:`~.ParameterExpression.__truediv__`
+   * - 4
+     - :meth:`~.ParameterExpression.__pow__`
+   * - 5
+     - :meth:`~.ParameterExpression.sin`
+   * - 6
+     - :meth:`~.ParameterExpression.cos`
+   * - 7
+     - :meth:`~.ParameterExpression.tan`
+   * - 8
+     - :meth:`~.ParameterExpression.arcsin`
+   * - 9
+     - :meth:`~.ParameterExpression.arccos`
+   * - 10
+     - :meth:`~.ParameterExpression.exp`
+   * - 11
+     - :meth:`~.ParameterExpression.log`
+   * - 12
+     - :meth:`~.ParameterExpression.sign`
+   * - 13
+     - :meth:`~.ParameterExpression.gradient`
+   * - 14
+     - :meth:`~.ParameterExpression.conjugate`
+   * - 15
+     - :meth:`~.ParameterExpression.subs`
+   * - 16
+     - :meth:`~.ParameterExpression.abs`
+   * - 17
+     - :meth:`~.ParameterExpression.arctan`
+   * - 255
+     - NULL
+
+The ``NULL`` value of 255 is only used to fill the op code field for
+entries that are not actual operations but indicate recursive definitions.
+Then the ``lhs_type`` and ``rhs_type`` fields are used to describe
+the operand types and can be one of the following utf8 encoded
+characters:
+
+.. list-table:: PARAM_EXPR_ELEM_V13 operand type values
+   :header-rows: 1
+
+   * - Value
+     - Type
+   * - ``n``
+     - ``None``
+   * - ``p``
+     - :class:`.Parameter`
+   * - ``f``
+     - ``float``
+   * - ``c``
+     - ``complex``
+   * - ``i``
+     - ``int``
+   * - ``s``
+     - Recursive :class:`.ParameterExpression` definition start
+   * - ``e``
+     - Recursive :class:`.ParameterExpression` definition stop
+   * - ``u``
+     - substitution
+
+if the type value is ``f`` or ``c`` the corresponding ``lhs`` or `rhs``
+field is the literal value with 0 padding for the full 128 bit width in case of a
+float. For ``i`` it's encoded as a 64 bit signed integer with 0 padding for the full
```

Comment:
I found the wording a bit confusing, I don't know if this is too verbose but I tried to reword this part a little up to my understanding:

```suggestion
If the type value is ``f`` ,``c`` or ``i``, the corresponding ``lhs`` or `rhs``
field widths are 128 bits each. In the case of floats, the literal value is encoded as a double 
with 0 padding, while complex numbers are encoded as real part followed by imaginary part,
taking up 64 bits each. For ``i`, the value is encoded as a 64 bit signed integer with 0 padding for the full
```